python/TestHarness/TestHarness.py

```diff
@@ -698,39 +710,58 @@ def handleTestResult(self, tester, output, result, start=0, end=0, add_to_table=
             print '\n\nFinal Test Results:\n' + ('-' * (TERM_COLS-1))
             for (test, output, result, timing, start, end) in sorted(self.test_table, key=lambda x: x[2], reverse=True):
                 print printResult(test, result, timing, start, end, self.options)
 
         time = clock() - self.start_time
+
         print '-' * (TERM_COLS-1)
-        print 'Ran %d tests in %.1f seconds' % (self.num_passed+self.num_failed, time)
 
-        if self.num_passed:
-            summary = '<g>%d passed</g>'
-        else:
-            summary = '<b>%d passed</b>'
-        summary += ', <b>%d skipped</b>'
-        if self.num_pending:
-            summary += ', <c>%d pending</c>'
-        else:
-            summary += ', <b>%d pending</b>'
-        if self.num_failed:
-            summary += ', <r>%d FAILED</r>'
+        # Print a different footer when performing a dry run
+        if self.options.dry_run:
+            print 'Processed %d tests in %.1f seconds' % (self.num_passed+self.num_skipped, time)
+            summary = '<b>%d would run</b>'
+            summary += ', <b>%d would be skipped</b>'
+
+            # Mask off TestHarness error codes to report parser errors
+            if self.error_code & Parser.getErrorCodeMask():
+                summary += ', <r>FATAL PARSER ERROR</r>'
+            if self.error_code & ~Parser.getErrorCodeMask():
+                summary += ', <r>FATAL TEST HARNESS ERROR</r>'
+
+            print colorText( summary % (self.num_passed, self.num_skipped),  "", html = True, \
+                             colored=self.options.colored, code=self.options.code )
+
         else:
-            summary += ', <b>%d failed</b>'
-
-        # Mask off TestHarness error codes to report parser errors
-        if self.error_code & Parser.getErrorCodeMask():
-            summary += ', <r>FATAL PARSER ERROR</r>'
-        if self.error_code & ~Parser.getErrorCodeMask():
-            summary += ', <r>FATAL TEST HARNESS ERROR</r>'
-
-        print colorText( summary % (self.num_passed, self.num_skipped, self.num_pending, self.num_failed),  "", html = True, \
-                         colored=self.options.colored, code=self.options.code )
-        if self.options.pbs:
-            print '\nYour PBS batch file:', self.options.pbs
+            print 'Ran %d tests in %.1f seconds' % (self.num_passed+self.num_failed, time)
+
+            if self.num_passed:
+                summary = '<g>%d passed</g>'
+            else:
+                summary = '<b>%d passed</b>'
+            summary += ', <b>%d skipped</b>'
+            if self.num_pending:
+                summary += ', <c>%d pending</c>'
+            else:
+                summary += ', <b>%d pending</b>'
+            if self.num_failed:
+                summary += ', <r>%d FAILED</r>'
+            else:
+                summary += ', <b>%d failed</b>'
+
+            # Mask off TestHarness error codes to report parser errors
```

Comment:
Looks to me like this block of code on down is identical to the one in the `if` statement just above. Move it outside the `if` statement to remove duplication.